src/providers/formattingProvider.utils.ts

```diff
@@ -316,7 +353,7 @@ export function purify(original: string): string {
     //
     let pure = cmdTrim
         .replace(/".*?"/g, '""') // replace string literals with empty string literal
-        .replace(/{.*}/g, '') // remove matching braces
+        .replaceAll(/{[^{}]*}/g, '') // remove matching braces
```

Comment:
I know my suggestion to use our own function instead of a custom `replaceAll` will make this code a bit uglier, but I think it's for the best overall

```ts
let pure = cmdTrim.replace(/".*?"/g, '""'); // replace string literals with empty string literal
pure = customReplaceAll(pure, /{[^{}]*}/g, ''); // remove matching braces
pure = pure.replace(/\s+/g, ' ') // collapse all spaces and tabs to single space
        .replace(/;.+/, '') // remove comments; must be last, semicolon may be inside string (expression)
        .trim();
```

Reply:
I save old function name `replaceAll`. Tests for this function are indirectly done in `purify` function tests (`purify` got regexp for `replaceAll`), I don't want duplicate regexps in code and tests.

Reply:
Since we're testing `purify`, we don't need custom tests for `replaceAll` :)